tests/course/test_for_review_course.py

```diff
@@ -0,0 +1,116 @@
+from django.core.validators import ValidationError
+from datetime import date
+
+
+@pytest.fixture
+def new_user():
+    user = User(username="test3",
+                first_name="test",
+                last_name="mctest",
+                email="test1@gmail.com")
+
+    user.set_password("PASSWORD")
+    return user
+
+
+@pytest.fixture
+def new_user_number_two():
+    user = User(username="natali",
+                first_name="shiraz",
+                last_name="yom tov",
+                email="shiraz@gmail.com")
+
+    user.set_password("PASSWORD2")
+    return user
+
+
+@pytest.fixture
+def new_teacher_course(persist_user):
+    teacher_course = TeacherCourses(course_id=1, teacher_id=persist_user,
+                                    course_name="Math", description="course", price=70)
+    return teacher_course
+
+
+@pytest.fixture
+def new_teacher_course_number_two(persist_user_number_two):
+    teacher_course = TeacherCourses(course_id=1, teacher_id=persist_user_number_two,
+                                    course_name="English", description="course", price=100)
+    return teacher_course
+
+
+@pytest.fixture
+def new_review(persist_user_number_two, persist_teacher_course):
+    review = Review(student_id=persist_user_number_two, course_id=persist_teacher_course,
+                    rating=1, content="Greate course", date=date(2023, 4, 17))
+    return review
+
+
+@pytest.fixture
+def new_review_two(persist_user, persist_teacher_course_number_two):
+    review = Review(student_id=persist_user, course_id=persist_teacher_course_number_two,
+                    rating=2, content="Greate course", date=date(2023, 4, 17))
+    return review
+
+
+@pytest.fixture
+def persist_user(new_user):
+    new_user.save()
+    return new_user
+
+
+@pytest.fixture
+def persist_user_number_two(new_user_number_two):
+    new_user_number_two.save()
+    return new_user_number_two
+
+
+@pytest.fixture
+def persist_teacher_course(new_teacher_course):
+    new_teacher_course.teacher_id.save()
+    new_teacher_course.save()
+    return new_teacher_course
+
+
+@pytest.fixture
+def persist_teacher_course_number_two(new_teacher_course_number_two):
+    new_teacher_course_number_two.teacher_id.save()
+    new_teacher_course_number_two.save()
+    return new_teacher_course_number_two
+
+
+@pytest.fixture
+def persist_review(new_review):
+    new_review.save()
+    return new_review
+
+
+@pytest.fixture
+def persist_review_number_two(new_review_two):
+    new_review_two.save()
+    return new_review_two
+
+
+@pytest.mark.django_db()
+class TestReview:
+    @pytest.mark.parametrize("rating", [0, 6])
+    def test_rating_limit(self, rating, persist_user, persist_teacher_course):
+        review = Review(student_id=persist_user, course_id=persist_teacher_course,
+                        rating=rating, content="Greate course", date=date(2023, 4, 17))
+        with pytest.raises(ValidationError):
+            review.full_clean()
```

Comment:
Maybe in this test, instead of creating a new Review object, we can take an existing fixture and modify the rating field.
I think its more clear what is being tested this way and removes fixtures that are not tied directly to the test
```suggestion
    def test_rating_limit(self, rating, new_review):
        new_review.rating = rating
        
        with pytest.raises(ValidationError):
            new_review.full_clean()
```

Reply:
You can also use the fixture from the contest instead of creating a new one

Reply:
> Maybe in this test, instead of creating a new Review object, we can take an existing fixture and modify the rating field. I think its more clear what is being tested this way and removes fixtures that are not tied directly to the test

Done